### ISSUE TITLE
terminal setup alternative

### ISSUE THREAD

**Issue by oxchronxo** [httptoolkit/httptoolkit#105]
When I try using the following to get the terminal setup I get the following error...
```
. <(curl -sS localhost:8001/setup)
(23) Failed writing body
```
Instead this worked for me...
```
eval "$(curl -sS localhost:8001/setup)"
```

Once I used this I could `printenv` and see all the added env vars.

**reason**: curl is shutting down the connection before the redirect is fully sourced.

**Comment by oxchronxo** [comment]
This happens consistently for me.

I believe this happens because curl sends an exit signal before the redirection has been completely read in by the `source` command. It's like having a book you're reading taken away right before you get to the end. Just a guess based on stderr.

**os**: Mac Catalina v10.15.6 (19G2021)
**curl**: 7.64.1 (x86_64-apple-darwin19.0) libcurl/7.64.1 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.39.2
**bash**: GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)

FWIW: This app has made my life so much easier. I don't have to wait on other people to update their apis for me to move forward with my work, I can fake it until THEY make it.

**Comment by pimterry** [comment]
With a bit more research (https://stackoverflow.com/questions/16703647/why-does-curl-return-error-23-failed-writing-body) it sounds like this error is from the curl end - i.e. curl is still writing the response, so probably hasn't closed anything, but its pipe to the `source` command which has been unexpectedly closed (so probably from the other end). Very weird!

I imagine this would happen if there was an error whilst running or parsing the received script though, which is definitely worth investigating.

What happens if you curl to a file, and then `source filename`? Does that work, or return any more error info? If it runs to completion (in any case, including via `eval`) it should print "HTTP Toolkit interception enabled" at the end.

**Comment by oxchronxo** [comment]
Just performing the curl operation succeeds. I'll try later with more verbose call to see what happens

**Comment by pimterry** [comment]
I think this is resolved - we did eventually (last year) move to the `eval` approach for Bash interception specifically, and as far as I'm aware that resolves this issue.

I'll close this now, but if anybody is still facing this, please do reply here or open a new issue to investigate further.
